libraries/chain/finality/qc.cpp

```diff
@@ -437,7 +437,14 @@ qc_vote_metrics_t aggregating_qc_t::vote_metrics(const qc_t& qc) const {
       size_t added = 0;
       for (size_t i = 0; i < votes.size(); ++i) {
          if (votes[i]) {
-            results.insert(finalizer_authority_ptr{finalizer_policy, &finalizer_policy->finalizers[i]}); // use aliasing shared_ptr constructor
+            results.insert(qc_vote_metrics_t::fin_auth{
+                  .fin_auth   = finalizer_authority_ptr{finalizer_policy, &finalizer_policy->finalizers[i]}, // use aliasing shared_ptr constructor
+                                                                                                           // add_policy_votes and add_votes in turn  is called on
+                                                                                                           // pending_finalizer_policy after on active_finalizer_policy.
+                                                                                                           // Therefore pending_finalizer_policy generation will be used
+                                                                                                           // for generation if the finalizer votes on both active and
+                                                                                                           // pending finalizer policies.
+                  .generation = finalizer_policy->generation});
```

Comment:
When we have long comments, I think it is preferable to not indent them too much. maybe start it on the same column as `.fin_auth`.

Reply:
Done by https://github.com/AntelopeIO/spring/pull/453/commits/42a361cd07a106647ab9582b73cc3247a3f5834f